src/rpm/signature/pgp.rs

```diff
@@ -98,14 +99,24 @@ impl Signer {
             PublicKeyAlgorithm::RSA => Ok(Self {
                 secret_key,
                 algorithm: AlgorithmType::RSA,
+                key_passphrase: None,
             }),
             PublicKeyAlgorithm::EdDSA => Ok(Self {
                 secret_key,
                 algorithm: AlgorithmType::EdDSA,
+                key_passphrase: None,
             }),
             algorithm => Err(Error::UnsupportedPGPKeyType(algorithm)),
         }
     }
+
+    /// Returns a new [Signer], with the PGP key passphrase provided.
```

Comment:
Technically it returns the same signer (while destroying the first copy via. move).  A better way of wording this might be.

`/// Configures the [Signer] with the provided PGP key passphrase`